taxonomy/utils.py

```diff
@@ -144,6 +190,72 @@ def refresh_course_skills(courses, should_commit_to_db):
     )
 
 
+def refresh_program_skills(programs, should_commit_to_db):
+    """
+    Refresh the skills associated with the provided programs.
+    """
+    all_failures = []
```

Comment:
nit: failures

Reply:
Already being used.

Reply:
You can rename the failure to program_skilll_failures. all_failures is semantically hard to understand.